src/Sparc.jjt

```diff
@@ -1433,6 +1436,83 @@ SimpleNode head() :
     jjtThis.beginColumn = n.beginColumn;
     return jjtThis;
   }
+| n = optimize_statement()
+  {
+    jjtThis.beginLine = n.beginLine;
+    jjtThis.beginColumn = n.beginColumn;
+    return jjtThis;
+  }
+}
+        
+SimpleNode optimize_statement() :
+{
+  Token t;
+}
+{
+  t = < MINIMIZE_OB > [ optimizeParameterList() ] < COLON > [ nonRelAtomList() ] < CB >
+  { 
+    jjtThis.image = t.image;
+    jjtThis.beginLine = t.beginLine;
+    jjtThis.beginColumn = t.beginColumn;
+    if (Settings.getSingletonInstance().getSolver() != ASPSolver.Clingo) 
+        throw new ParseException("line " + jjtThis.beginLine + ", column " + jjtThis.beginColumn + ": optimization statements are not supported in DLV");
+    return jjtThis;
+  }
+| t = < MAXIMIZE_OB > [ optimizeParameterList() ] < COLON > [ nonRelAtomList() ] < CB >
+  { 
+    jjtThis.image = t.image;
+    jjtThis.beginLine = t.beginLine;
+    jjtThis.beginColumn = t.beginColumn;
+    if (Settings.getSingletonInstance().getSolver() != ASPSolver.Clingo) 
+        throw new ParseException("line " + jjtThis.beginLine + ", column " + jjtThis.beginColumn + ": optimization statements are not supported in DLV");
+    return jjtThis;
+  }
+}
+
+void optimizeParameterList() :
+{}
+{
+    optimizeParameter()
+    (
+    < COMMA > optimizeParameter()
+    )*
+}
+
+void nonRelAtomList() :
+{}
+{
+    nonRelAtom()
+    (
+    < COMMA > nonRelAtom()
+    )*
+}
+        
+void optimizeParameter() :

```

Comment:
Do we need this optimizeParameter to be so complicated? I thought we agreed that it can be just a variable for now. Allowing more is not always a good thing, unless there is a reason for it.